core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/accesscontrol/resourcemapper/AccessControlledResourceMapper.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.core.accesscontrol.resourcemapper;
+
+import com.expediagroup.streamplatform.streamregistry.accesscontrol.domain.AccessControlledResource;
+
+public interface AccessControlledResourceMapper {
```

Comment:
can't we introduce generic support here?, so implementation doesn't need an explicit cast when calling `map()`

```
public interface AccessControlledResourceMapper<T> {

      boolean canMap(Object domainObject);

      AccessControlledResource map(T domainObject);
}